actionpack/test/dispatch/request/session_test.rb

```diff
@@ -103,6 +103,19 @@ def test_fetch
         end
       end
 
+      def test_with_indifferent_access
+        env = {}
+        s = Session.create(store, env, {})
+
+        s[:one] = { test: "deep" }
+        s[:two] = { "test" => "deep" }
+
+        s2 = Session.create(store, env, {})
```

Comment:
@tomprats Weird. By just using your test and asserting `s`, it fails for me. So you wouldn't need `s2`. What am I missing?